share/gpodder/extensions/10_youtube-dl.py

```diff
@@ -357,10 +359,22 @@ def fetch_info(self, url, tempname, reporthook):
 
         # Need the proxy_url from src/gpodder/config.py:get_proxies_from_config()
         if gpodder.config._proxies:
-            opts['proxy'] = gpodder.config._proxies['http']
-            logger.debug(f"Setting proxy from network setting proxy: {opts['proxy']}")
+            fetch_opts['proxy'] = gpodder.config._proxies['http']
+            logger.debug(f"Setting proxy from network setting proxy: {fetch_opts['proxy']}")
+
+        opts = self._ydl_opts.copy()
+        # Update or extend existing dict and list values
```

Comment:
Why is this needed when the other two sites could get away with something like `opts.update(fetch_opts)`?

Reply:
Thank you, now they use the same syntax as the conditional keys. We can now move the `cachedir` lines together and group them between full line comments before keys instead of EOL comments after some of them.

Give me a few days to test it.